cli/index.js

```diff
@@ -39,13 +43,354 @@ const signUp = (args) => {
+        });
+}
+
+//* updated user 
+const updateUser = (args) => {
+
+    const { password, confirmPassword, username, firstName, 
+    lastName, email, bio, website, location, picture, birthday } = args;
+
+    const updateUserData = {
+        password,
+        confirmPassword,
+        username,
+        firstName,
+        lastName,
+        email,
+        bio,
+        website,
+        location,
+        picture,
+        birthday
+    };
+
+    console.log(chalk.yellow.bold("updating..."));
+
+    axios
+        .put("/user/{username}", updateUserData)
+        .then((res) => {
+            console.log(res.data);
+
+            const FBIdToken = `Bearer ${res.data.token}`;
+            // TODO: create a file in the user's machine to store the auth token
+            console.log(chalk.green.bold(" update successfully ✔️"));
+        })
+        .catch((error) => {
+            console.log(chalk.red.bold(`Could not update this account -> code: ${error.response.statusText}, message: ${JSON.stringify(error.response.data)}`));
+        });
+}
+//get post command definition
+const getPost = (args) => {
+
+    console.log(chalk.yellow.bold("getting post..."));
+
+    const { postId } = args;
+    
+        const getPostData = {
+            postId
+        };
+
+    axios
+        .get("/posts", getPostData)
+        .then((res) => {
+            console.log(res.data);
+
+            const FBIdToken = `Bearer ${res.data.token}`;
+            // TODO: create a file in the user's machine to store the auth token
+            console.log(chalk.green.bold("you got the post successfully ✔️"));
+        })
+        .catch((error) => {
+            console.log(chalk.red.bold(`Could not obtain the post -> code: ${error.response.statusText}, message: ${JSON.stringify(error.response.data)}`));
+        });
+}
+//create post command definition
+const createPost = (args) => {
+
+    const { post, canLike, canUnlike, likes, tag } = args;
+
+    const postData = {
+        post,
+        canLike,
+        canUnlike,
+        likes,
+        tag
+    };
+
+    console.log(chalk.yellow.bold("posting..."));
+
+    axios
+        .post("/posts", postData)
+        .then((res) => {
+            console.log(res.data);
+
+            const FBIdToken = `Bearer ${res.data.token}`;
+            // TODO: create a file in the user's machine to store the auth token
+            console.log(chalk.green.bold("post created successfully ✔️"));
+        })
+        .catch((error) => {
+            console.log(chalk.red.bold(`Could not create the post -> code: ${error.response.statusText}, message: ${JSON.stringify(error.response.data)}`));
+        });
+}
+//update post command definition
+const updatePost = (args) => {
+
+    const { postId, canLike, canUnlike, likes, tag } = args;
+
+    const putPostData = {
+        postId,
+        canLike,

```

Comment:
you cannot update the postId, otherwise it wouldn't be an ID if it can be updated.